discord_bot/discord_bot.py

```diff
@@ -51,28 +52,26 @@ def load_config(config_file_path):
         with open(config_file_path) as f:
             return json.load(f)
 
-    def get_gateway(self):
+    async def get_gateway(self):
         """
         Caches a gateway value, authenticates, and retrieves a new URL
         :return: gateway URL
         """
-        r = requests.get("{}/gateway".format(self.config["discord_api_endpoint"]))
-        assert 200 == r.status_code, r.reason
-        return r.json()["url"]
+        with ClientSession() as session:
```

Comment:
Does this need to be `async with ClientSession() as session:`? I'm not sure myself, see what the internet says.